File: problems/3107_minimum_operations_to_make_median_of_array_equal_to_k/Solution.java
import java.util.*;

class Solution {
    public long minOperationsToMakeMedianK(int[] nums, int k) {
        Arrays.sort(nums);
        int m = nums.length / 2;
        long ops = 0;
        for (int i = 0; i < nums.length; i++) {
            if (i <= m && nums[i] > k) ops += nums[i] - k;
            else if (i >= m && nums[i] < k) ops += k - nums[i];
        }
        return ops;
    }
}

File: problems/3107_minimum_operations_to_make_median_of_array_equal_to_k/solution.py
from typing import List


class Solution:
    def minOperationsToMakeMedianK(self, nums: List[int], k: int) -> int:
        nums.sort()
        n = len(nums)
        m = n // 2
        ops = 0
        for i in range(n):
            if i <= m and nums[i] > k:
                ops += nums[i] - k
            elif i >= m and nums[i] < k:
                ops += k - nums[i]
        return ops
